aiida/orm/implementation/entities.py

```diff
@@ -0,0 +1,354 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Classes and methods for backend non-specific entities"""
+
+import abc
+
+from aiida.orm.utils.node import clean_value, validate_attribute_extra_key
```

Comment:
I am thinking, since `validate_attribute_extra_key` is now really only used in the backend implementation, maybe it makes sense to move it in this package, so move it into module `aiida.orm.implementation.utils`. The same would almost go for `clean_value`. It is just used by some modules in `aiida.tools`. However, since `aiida.orm.utils.node` is not public facing API, we can just move both. Up to our discretion. What do you think?

Reply:
I agree. It's not really placed appropriately in `aiida.orm.utils.node`, and moving it shouldn't affect the user. I'll make a new module `aiida.orm.implementation.utils` and move the methods there.

Reply:
I think you did this so we can mark this as resolved